src/irmin-pack/layered/checks.ml

```diff
@@ -82,11 +79,15 @@ struct
     type t = { hash_size : Layer_stat.size; files : files_layer }
     [@@deriving irmin]
 
+    let v ~root =
+      let version = Layer_stat.detect_version ~root in
+      Layer_stat.v ~version ~root
+
     let v ~root =
       read_flip ~root >|= fun flip ->
-      let lower = Layer_stat.v ~root:(Layout.lower ~root)
-      and upper1 = Layer_stat.v ~root:(Layout.upper1 ~root)
-      and upper0 = Layer_stat.v ~root:(Layout.upper0 ~root) in
+      let lower = v ~root:(Layout.lower ~root)
+      and upper1 = v ~root:(Layout.upper1 ~root)
+      and upper0 = v ~root:(Layout.upper0 ~root) in
```

Comment:
I suspect we want to enforce that all IOs have the same version, since that's a possible integrity failure?